### ISSUE TITLE
Adding support for FABA audiocube (ita/fra/spa)

### ISSUE THREAD

**Issue by tintifaxe** [oyooyo/audiocube#22]
Thanks for the repo! Handy stuff which makes a lot of kids and parents happy :)
I can bring Information about the FABA box from the same manufacturer as the other boxes. Unfortunately, I haven't managed to decrypt the files yet.

**Comment by tintifaxe** [comment]
I just successfully decrypted the mp3 files on the FABA box.
 It is a Bit rotation +3 followed by 4 byte XOR  (similar to the Lidl Box)
 next step is to encrypt own mp3 files

**Comment by drego85** [comment]
Hi [tintifaxe](https://github.com/tintifaxe),
can you please provide me more details about the XOR rotation you exploited so I can update the code in this repository?
What is the XOR key that you used?
Thanks

**Comment by drego85** [comment]
While waiting for @tintifaxe's response, @oyooyo may I ask you for some advice? 
How were the XOR keys from the other boxes found? Firmware dump? 
So I can also try to extract the key and collaborate on this project for the Faba Box.
Thanks

**Comment by tintifaxe** [comment]
Here the Key for the box:

[ALGORITHM]
NAME=Fbox
CHANGEFILEEXT=1
OPERATIONSCOUNT=2
SOURCEEXT=.MKI
DESTEXT=.MP3
CHANGEID3=0
[OPERATION0]
TYPE=1
ROTATEDIRECTION=1
ROTATECOUNT=3
XORKEY=
[OPERATION1]
TYPE=0
ROTATEDIRECTION=0
ROTATECOUNT=0
XORKEY=0x06,0xA0,0x23,0x04

**Comment by vinpanasiti** [comment]
Hello, have you tried to convert an MP3 to MKI with this key?
I can convert MKI->MP3 but not work for MP3->MKI

**Comment by tintifaxe** [comment]
Yes, I have successfully used the key backward with Smp2MP3. I don't have the box at hand, so I cant tell you nymore how I configured the encryption algorithm to encrypt MP3->MKI. Try XOR first and then Bit rotation to the left.

I guess you want to assign your RFID tags to your mp3 files.

![Image](https://github.com/user-attachments/assets/fd6280ef-e1b7-45c5-81cb-017cb9ebc65e)

**Comment by vinpanasiti** [comment]
Yes, exactly. I tried to do the conversion and add the track to the end of a playlist linked to an RFID tag, but when it gets to that track, it stays silent. 

Have you figured out how to program a new tag to activate an entire playlist that I’ve created? My goal is to create a custom playlist of songs that can be triggered with a new tag.

Thank you very much!!!

**Comment by Lorenzius85** [comment]
> Here the Key for the box:
> 
> [ALGORITHM] NAME=Fbox CHANGEFILEEXT=1 OPERATIONSCOUNT=2 SOURCEEXT=.MKI DESTEXT=.MP3 CHANGEID3=0 [OPERATION0] TYPE=1 ROTATEDIRECTION=1 ROTATECOUNT=3 XORKEY= [OPERATION1] TYPE=0 ROTATEDIRECTION=0 ROTATECOUNT=0 XORKEY=0x06,0xA0,0x23,0x04

Hi, I'm trying to convert .mp3->.mki files but this algorithm only allows .mki->.mp3 conversion. I tried to put XOR first and then reverse rotation but it didn't work. Could you help me to fix the algorithm?

**Comment by mangioacasa** [comment]
> Sì, esattamente. Ho provato a fare la conversione e ad aggiungere la traccia alla fine di una playlist collegata a un tag RFID, ma quando arriva a quella traccia, rimane in silenzio.
> 
> Hai capito come programmare un nuovo tag per attivare un'intera playlist che ho creato? Il mio obiettivo è creare una playlist personalizzata di brani che possono essere attivati ​​con un nuovo tag.
> 
> Grazie mille!!!

before converting the mp3 file you have to change the song title with the folder name followed by the song number/name.
example K0091CP03

**Comment by mangioacasa** [comment]
> Yes, exactly. I tried to do the conversion and add the track to the end of a playlist linked to an RFID tag, but when it gets to that track, it stays silent.
> 
> Have you figured out how to program a new tag to activate an entire playlist that I’ve created? My goal is to create a custom playlist of songs that can be triggered with a new tag.
> 
> Thank you very much!!!

before converting the mp3 file you have to change the song title with the folder name followed by the song number/name.
example K0091CP03

**Comment by fedezinho** [comment]
> Yes, exactly. I tried to do the conversion and add the track to the end of a playlist linked to an RFID tag, but when it gets to that track, it stays silent.
> 
> Have you figured out how to program a new tag to activate an entire playlist that I’ve created? My goal is to create a custom playlist of songs that can be triggered with a new tag.
> 
> Thank you very much!!!

before converting the mp3 file you have to change the song title with the folder name followed by the song number/name.
example K0091CP03

**Comment by Zarhkoh** [comment]
Hello everyone, I am searching for the Fairy tales and Mr. Men & Little Miss in english, italian & German in SMP or MP3 versions. If anyone have them, can you share a link ? 👀 
Thank you very much !!
